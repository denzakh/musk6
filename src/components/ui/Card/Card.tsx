import { FC } from 'react'
import { CardType } from '../../../typos'

interface CardProps {
  card: CardType
  onClick: () => void
  isInCart: boolean
}

export const Card: FC<CardProps> = ({ card, onClick, isInCart }) => {
  return (
    <div className="card bg-base-100  shadow-sm">
      <figure>
        <img src={card?.img} alt={card?.title} />
      </figure>
      <div className="card-body">
        <h2 className="card-title">{card?.title}</h2>
        <p>{card?.text}</p>
        <div className="card-actions mt-3 flex justify-between items-center">
          <div className="card-title text-accent">{card?.price} €</div>
          <button
            className={`btn ${isInCart ? 'btn-accent' : 'btn-primary'}`}
            onClick={onClick}
            data-testid={`addCartBtn${card.id}`}
          >
            {isInCart ? 'En carrito' : 'Añadir al carrito'}
          </button>
        </div>
      </div>
    </div>
  )
}
