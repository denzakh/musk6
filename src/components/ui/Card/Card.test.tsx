import { Card } from './Card'
import { describe, it, expect, vi } from 'vitest'
import { render, fireEvent, screen, cleanup } from '@testing-library/react'

const card = {
  id: 1,
  title: 'Boba Fett',
  price: 80,
  img: 'https://vieraboschkova.github.io/swapi-gallery/static/assets/img/people/22.jpg',
  text: 'Figura de Boba Fett de la serie Elite de Star Wars, 6" de metal fundido a presión, nueva, envío gratis',
}

describe('Card', async () => {
  it('El artículo de la tarjeta no está en el carrito.', async () => {
    render(<Card key={card.title} card={card} onClick={() => {}} isInCart={false} />)

    expect(screen.queryByText('Añadir al carrito')).toBeTruthy()
    expect(screen.queryByText('En carrito')).toBeNull()

    cleanup()
  })

  it('Snapshot: El artículo de la tarjeta no está en el carrito.', async () => {
    const comp = render(<Card key={card.title} card={card} onClick={() => {}} isInCart={false} />)
    expect(comp).toMatchSnapshot()
    cleanup()
  })

  it('Añadiendo el artículo 1 al carrito', async () => {
    render(<Card key={card.title} card={card} onClick={() => {}} isInCart={true} />)

    expect(screen.queryByText('Añadir al carrito')).toBeNull()
    expect(screen.queryByText('En carrito')).toBeTruthy()
    cleanup()
  })

  it('Añadiendo el artículo 1 al carrito', async () => {
    const handleCart = vi.fn(() => 1)

    render(<Card key={card.title} card={card} onClick={() => handleCart()} isInCart={false} />)

    const addCartBtn1Element = await screen.findByTestId('addCartBtn1')
    fireEvent.click(addCartBtn1Element)
    expect(handleCart).toHaveBeenCalled()
  })
})
