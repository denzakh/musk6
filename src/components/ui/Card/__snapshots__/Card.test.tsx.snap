// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Card > Snapshot: El artículo de la tarjeta no está en el carrito. 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="card bg-base-100  shadow-sm"
      >
        <figure>
          <img
            alt="Boba Fett"
            src="https://vieraboschkova.github.io/swapi-gallery/static/assets/img/people/22.jpg"
          />
        </figure>
        <div
          class="card-body"
        >
          <h2
            class="card-title"
          >
            Boba Fett
          </h2>
          <p>
            Figura de Boba Fett de la serie Elite de Star Wars, 6" de metal fundido a presión, nueva, envío gratis
          </p>
          <div
            class="card-actions mt-3 flex justify-between items-center"
          >
            <div
              class="card-title text-accent"
            >
              80
               €
            </div>
            <button
              class="btn btn-primary"
              data-testid="addCartBtn1"
            >
              Añadir al carrito
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="card bg-base-100  shadow-sm"
    >
      <figure>
        <img
          alt="Boba Fett"
          src="https://vieraboschkova.github.io/swapi-gallery/static/assets/img/people/22.jpg"
        />
      </figure>
      <div
        class="card-body"
      >
        <h2
          class="card-title"
        >
          Boba Fett
        </h2>
        <p>
          Figura de Boba Fett de la serie Elite de Star Wars, 6" de metal fundido a presión, nueva, envío gratis
        </p>
        <div
          class="card-actions mt-3 flex justify-between items-center"
        >
          <div
            class="card-title text-accent"
          >
            80
             €
          </div>
          <button
            class="btn btn-primary"
            data-testid="addCartBtn1"
          >
            Añadir al carrito
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
